### PR TITLE
configurable support for symlinks

### PR DESCRIPTION
**Pull request by ScottDillman**
Was recently shopping around for a good scaffolding tool for projects and found kickstart, the only feature missing for me was the ability to follow symlinks so I can share common files between templates. This simple PR adds the ability to follow symlinks as a template config parameter but does not change the default behavior.

### CODE REVIEW

File: src/generation.rs [hunk 1]
```diff
@@ -95,7 +95,7 @@ impl Template {
         };
 
         // And now generate the files in the output dir given
-        let walker = WalkDir::new(&start_path)
+        let walker = WalkDir::new(&start_path).follow_links(definition.symlinks)
```
Comment (Keats) [inline src/generation.rs:98]: Should we follow them by default?

File: src/definition.rs [hunk 2]
```diff
@@ -200,6 +203,7 @@ mod tests {
             name = "Test template"
             description = "A description"
             kickstart_version = 1
+            symlinks = false
```
Comment (Keats) [inline src/definition.rs:206]: If we use an option, I'd rather have it `follow_symlinks`

File: src/generation.rs (hunk 1 above)
Comment (ScottDillman) [reply src/generation.rs:98]: I wasn't sure of the original intent so I left it disabled by default, but I'm all for having it be the default case as long as it handles cycles during the walk and it doesn't effect Windows users. I'll test and makes sure that it doesn't introduce any issues there and if not make it the default.

File: src/definition.rs (hunk 2 above)
Comment (ScottDillman) [reply src/definition.rs:206]: much more descriptive, agreed. I'll make that change.

File: src/generation.rs (hunk 1 above)
Comment (Keats) [reply src/generation.rs:98]: Initially the use case was for templates to be self contained, eg a git repo so there was no real need for symlinks

**Pull request merged**